src/jit/compiler.h

```diff
@@ -297,6 +297,208 @@ enum RefCountState
     RCS_NORMAL,  // normal ref counts (from lvaMarkRefs onward)
 };
 
+//--------------------------------------------
+//
+// VariableLiveRange: Represent part of the life of a variable. A
+//      variable lives in a location (represented with struct "siVarLoc")
+//      between two assembly offsets.
+//
+// Notes:
+//    We use emitLocation and not NATTIVE_OFFSET because location
+//    is captured when code is being generated (genCodeForBBList
+//    and genGeneratePrologsAndEpilogs) but only once the whole
+//    methodit's code is done the NATTIVE_OFFSET won't change.
+//    It is also IL_OFFSET, but this is more accurate and the
+//    debugger is expeecting assembly offsets.
+//    This class doesn't have behaviour attached to itself, it is
+//    just putting a name to a representation. It is used to build
+//    typedefs LiveRangeList and LiveRangeListIterator, which are
+//    basically a list of this class and a const_iterator of that
+//    list.
+//
+class VariableLiveRange
+{
+public:
+    emitLocation               m_StartEmitLocation; // first position from where "m_VarHome" becomes valid
+    emitLocation               m_EndEmitLocation;   // last position where "m_VarHome" is valid
+    CodeGenInterface::siVarLoc m_VarHome;           // variable home
+
+    VariableLiveRange(CodeGenInterface::siVarLoc varHome, emitLocation startEmitLocation, emitLocation endEmitLocation)
+        : m_StartEmitLocation(startEmitLocation), m_EndEmitLocation(endEmitLocation), m_VarHome(varHome)
+    {
+    }
+};
+
+typedef jitstd::list<VariableLiveRange> LiveRangeList;
```

Comment:
I have reservations about using `jitstd::list` for this. It's a doubly linked list and so far I haven't found anything in this code that requires backward traversal. Unfortunatelly we don't have a `jitstd::forward_list`, maybe we should add that one day.

Reply:
And it's not only that list nodes have an extra pointer in them but the list object itself is pretty fat, considering that you need one for each IL local variable - 5 pointers. And 2 of them are allocators...

Reply:
Can we start using std structs as std::forward_list https://en.cppreference.com/w/cpp/container/forward_list ?

Reply:
No, the real `std` namespace is not usable in the JIT at this time. Only what's in `jitstd`.

Reply:
Should we open an Issue for this?

Reply:
I don't think it's worthwhile, especially considering the high numer of opened issues the JIT already has. If someone needs `forward_list` and feels that it's worth adding it then they can just add it, no need for an issue. Anyway, it turns out that `forward_list` can't help here, it's actually a stack and you'd need to reverse it to traverse it in the correct order.

You'd be better of just not using any container class and instead manually link variable ranges in a singly linked list. I don't usually recommend that but when you have only one place where you need to add nodes to the list that becomes reasonable.

Reply:
I need essentially the push_back(), back(), size() operations (better if they have constant time complexity) and a sort of iterator to traverse the whole list. If I create my own nodes supporting this staff, I will be describing a forward_list with other name.

Reply:
>  I will be describing a forward_list with other name

No because `forward_list` unfortunately works like a stack. It can only be traversed starting from the most recent added node but you need to do it the other way around.

Reply:
Sorry, you are right, I was thinking in the concept of a single linked list without thinking about the order.

Reply:
@BrianBohe I think at this point you should focus on ensuring the functionality is complete, and worry about optimizing the representation afterwards. It's reasonable to open issues related to this feature that can be addressed after the first merge.